config/config.go

```diff
@@ -1372,6 +1372,7 @@ func getGoGCEnv() int {
 // OTLPConfig is the configuration for writing to the OTLP endpoint.
 type OTLPConfig struct {
 	PromoteResourceAttributes []string `yaml:"promote_resource_attributes,omitempty"`
+	AllowUTF8                 bool     `yaml:"allow_utf8,omitempty"`
```

Comment:
This is still a bool, is there a reason you don't want to go with an enum?  The implementation can collapse to bool for now and we can refactor for enum later if needed.

https://github.com/prometheus/prometheus/pull/14974#discussion_r1775805810

https://github.com/prometheus/prometheus/pull/14974#issuecomment-2379444231